### PR TITLE
LibJS: Fix some module related issues

### PR DESCRIPTION
**Pull request by davidot**
For the spec issue see also https://github.com/tc39/ecma262/issues/2823 , which I'll try to fix myself later

The last commit fixes a recent regression of a test262 test

### CODE REVIEW

File: Userland/Libraries/LibJS/CyclicModule.cpp [hunk 1]
```diff
@@ -191,10 +191,23 @@ ThrowCompletionOr<Promise*> CyclicModule::evaluate(VM& vm)
     // 2. Assert: module.[[Status]] is linked, evaluating-async, or evaluated.
     VERIFY(m_status == ModuleStatus::Linked || m_status == ModuleStatus::EvaluatingAsync || m_status == ModuleStatus::Evaluated);
 
+    // NOTE: The spec does not catch the case where evaluate is called twice on a script which failed
+    //       during evaluation. This means the script is evaluated but does not have a cycle root.
+    //       In that case we first check if this module itself has a top level capability.
+    //       See also: https://github.com/tc39/ecma262/issues/2823 .
+
+    // If module.[[TopLevelCapability]] is not empty, then
```
Comment (linusg) [inline Userland/Libraries/LibJS/CyclicModule.cpp:199]: Missing step number

File: Userland/Libraries/LibJS/AST.h [hunk 2]
```diff
@@ -54,6 +54,7 @@ class ASTNode : public RefCounted<ASTNode> {
     virtual void dump(int indent) const;
 
     [[nodiscard]] SourceRange source_range() const;
+    u32 start_offset_in_source() const { return m_start_offset; }
```
Comment (linusg) [inline Userland/Libraries/LibJS/AST.h:57]: pure getters must match member names

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/CyclicModule.cpp (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibJS/CyclicModule.cpp:199]: This (currently) has no spec number, should I just put `?.` for now? Or leave out the spec comments

Comment (linusg) [reply Userland/Libraries/LibJS/CyclicModule.cpp:199]: if it has a comment, it must have a number :)
Either give it the same as where you copied it from, or omit - up to you

**Pull request merged**